braindecode/augmentation/functional.py

```diff
@@ -969,3 +970,115 @@ def mixup(X, y, lam, idx_perm):
         y_b[idx] = y[idx_perm[idx]]
 
     return X_mix, (y_a, y_b, lam)
+
+
+def _segmentation_reconstruction(X, n_segments, rng):
+    n_trials, n_channels, window_size = X.shape
+    # Segment Size
+    segment_size = window_size // n_segments
+    # Initialize an empty tensor for augmented data
+    X_aug = torch.zeros_like(X)
+    # Generate random indices within the class-specific dataset
+    rand_idx = rng.randint(0, n_trials, (n_trials, n_segments))
```

Comment:
Hi @cedricrommel,

I was studying this random samplings and I think the only way to move it to transforms.py would copy practically the whole functional in transforms.py, since the samplings envolves iterations over the batch (analogous problem in GaussianNoise, in which we would have to move all function transforms.py also). If you have a suggestion that I'm not seeing, please, send me a message so I can fix it!

Reply:
Perfect ! That’s precisely what I had in mind :)